caravel/sql_lab.py

```diff
@@ -114,49 +114,123 @@ def handle_error(msg):
             time.sleep(1)
             polled = cursor.poll()
 
-    columns = None
+    column_names = None
     data = None
     if result_proxy.cursor:
-        columns = [col[0] for col in result_proxy.cursor.description]
+        column_names = [col[0] for col in result_proxy.cursor.description]
         data = result_proxy.fetchall()
-        df = pd.DataFrame(data, columns=columns)
-        df = df.where((pd.notnull(df)), None)
-        # TODO consider generating tuples instead of dicts to send
-        # less data through the wire. The command bellow does that,
-        # but we'd need to align on the client side.
-        # data = df.values.tolist()
-        data = df.to_dict(orient='records')
+    df = pd.DataFrame(data, columns=column_names)
+    df = df.where((pd.notnull(df)), None)
+    # TODO consider generating tuples instead of dicts to send
+    # less data through the wire. The command bellow does that,
+    # but we'd need to align on the client side.
+    # data = df.values.tolist()
+    data = df.to_dict(orient='records')
 
     query.rows = result_proxy.rowcount
     query.progress = 100
     query.status = QueryStatus.SUCCESS
     if query.rows == -1 and data:
         # Presto doesn't provide result_proxy.row_count
         query.rows = len(data)
-
-    # CTAs queries result in 1 cell having the # of the added rows.
     if query.select_as_cta:
         query.select_sql = '{}'.format(database.select_star(
             query.tmp_table_name, limit=query.limit))
-
     query.end_time = utils.now_as_float()
     session.commit()
 
-    payload = {
-        'query_id': query.id,
-        'status': query.status,
-        'data': [],
-    }
-    if query.status == models.QueryStatus.SUCCESS:
-        payload['data'] = data
-        payload['columns'] = columns
-    else:
-        payload['error'] = query.error_message
     if return_results:
+        payload = {
+            'query_id': query.id,
+            'status': query.status,
+            'data': [],
+        }
+        if query.status == models.QueryStatus.SUCCESS:
+            payload['data'] = data
+            payload['columns'] = get_columns_dict(df) if column_names else None
+        else:
+            payload['error'] = query.error_message
         return payload
     '''
     # Hack testing using a kv store for results
     key = "query_id={}".format(query.id)
     logging.info("Storing results in key=[{}]".format(key))
     cache.set(key, json.dumps(payload, default=utils.json_iso_dttm_ser))
     '''
+
+
+THRESHOLD = 95
+SAMPLE_SIZE = 100
+
+
+# TODO(bkyryliuk): add support for the conventions like: *_dim or dim_*
+#     - dimensions, *_ts, ts_*, ds_*, *_ds - datetime, etc.
+# TODO(bkyryliuk): recognize integer encoded enums.
+def get_columns_dict(df):
```

Comment:
short simple docstring on 10+ command functions, todos can be moved inside the docstring